pkg/repository/mirror.go

```diff
@@ -248,18 +249,46 @@ func (l *httpMirror) prepareURL(resource string) string {
 	return url
 }
 
+func (l *httpMirror) isRetryable(err error) bool {
+	retryableList := []string{
+		"unexpected EOF",
+		"stream error",
+		"server returned 502 Bad Gateway",
+	}
+
+	for _, text := range retryableList {
+		if strings.Contains(strings.ToLower(err.Error()), strings.ToLower(text)) {
+			return true
+		}
+	}
+	return false
+}
+
 // Download implements the Mirror interface
 func (l *httpMirror) Download(resource, targetDir string) error {
 	tmpFilePath := filepath.Join(l.tmpDir, resource)
 	dstFilePath := filepath.Join(targetDir, resource)
 	// downloaded file is stored in a temp directory and the temp directory is
 	// deleted at Close(), in this way an interrupted download won't remain
 	// any partial file on the disk
-	r, err := l.download(l.prepareURL(resource), tmpFilePath, 0)
+	var err error
+	_ = utils.Retry(func() error {
```

Comment:
I think we should limit the retry count and retry duration.

Reply:
The utils.Retry has default value for that:
```golang
func Retry(doFunc func() error, opts ...RetryOption) error {
	var cfg RetryOption
	if len(opts) > 0 {
		cfg = opts[0]
	} else {
		cfg = RetryOption{
			Attempts: defaultAttempts,
			Delay:    defaultDelay,
			Timeout:  defaultTimeout,
		}
	}
```